# Analysis of NBA dataset

### Ziah D.

10/18/19


## Abstract

For this lab I am using pandas in order to, read the NBA dataset, and then group together the specific data for a certain team. From this I plan on analyzing the quartile ranges, along with the mean, max, and minimum.

## Dataset Preperation

I got this dataset from Ms. Sconyers. The dataset has a total of 458 rows and 9 columns, measuring data taken from various nba teams. 

I am writing the code below in order to open the file, close th file and then read the data. This important because it allows us to then jump right in to using the data to then analyze it.

In [21]:
# Import all the modules I want to use in this homework, so they are visible to my notebook.Makes pandas accesible and shortens it to pd.
import pandas as pd


In [22]:
#creating the dataframe
my_df = pd.read_csv("nba.csv")


## Data Modeling

Now that I have all of the information that I need I am using the following pieces of code in order to look at what the data looks like. The first line of code that I asked for calls the first five rows of infomation. Afterwards, in another line of code, I ask for the last three lines of information. Having both the first and last lines allow me to see where the code starts and what the last rows look like/how many rows there are in the dataset.

In [23]:
my_df.head(5)

Name            Team  Number Position  Age  Height  Weight  \
0  Avery Bradley  Boston Celtics       0       PG   25   2-Jun     180   
1    Jae Crowder  Boston Celtics      99       SF   25   6-Jun     235   
2   John Holland  Boston Celtics      30       SG   27   5-Jun     205   
3    R.J. Hunter  Boston Celtics      28       SG   22   5-Jun     185   
4  Jonas Jerebko  Boston Celtics       8       PF   29  10-Jun     231   

             College     Salary  
0              Texas  7730337.0  
1          Marquette  6796117.0  
2  Boston University        NaN  
3      Georgia State  1148640.0  
4                NaN  5000000.0

In [24]:
my_df.tail(3)

Name       Team  Number Position  Age  Height  Weight College  \
454     Raul Neto  Utah Jazz      25       PG   24   1-Jun     179     NaN   
455  Tibor Pleiss  Utah Jazz      21        C   26   3-Jul     256     NaN   
456   Jeff Withey  Utah Jazz      24        C   26  Jul-00     231  Kansas   

        Salary  
454   900000.0  
455  2900000.0  
456   947276.0

After reading the first and last  lines of the data, I then move on to using our best friend info. Info allows me, as a coder, to look at what types of data there are in the dataset, and if there is any missing data.

In [25]:
my_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 457 entries, 0 to 456
Data columns (total 9 columns):
Name        457 non-null object
Team        457 non-null object
Number      457 non-null int64
Position    457 non-null object
Age         457 non-null int64
Height      457 non-null object
Weight      457 non-null int64
College     373 non-null object
Salary      446 non-null float64
dtypes: float64(1), int64(3), object(5)
memory usage: 32.2+ KB


After using info, our best frined, we move on to our second best friend, descibe. Describe shows a lot of the ranges in the data. This is imortant becuase in previous labs we had to calculate each of these numbers individaully. 

In [26]:
my_df.describe()

/Users/ziahgraceolivadaily/anaconda/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


Number         Age      Weight        Salary
count  457.000000  457.000000  457.000000  4.460000e+02
mean    17.678337   26.938731  221.522976  4.842684e+06
std     15.966090    4.404016   26.368343  5.229238e+06
min      0.000000   19.000000  161.000000  3.088800e+04
25%      5.000000   24.000000  200.000000           NaN
50%     13.000000   26.000000  220.000000           NaN
75%     25.000000   30.000000  240.000000           NaN
max     99.000000   40.000000  307.000000  2.500000e+07

The code below is key. Groupby allows us to single out the data that you want and then organizes that data. 

In [27]:
#applying groupby() function to 
#group the data on team value
# the groupby is group all the teams...
gk = my_df.groupby('Team')

#Lets print the first entries
#in all the groups formed
gk.first()

Name  Number Position  Age  Height  \
Team                                                                          
Atlanta Hawks                   Kent Bazemore      24       SF   26   5-Jun   
Boston Celtics                  Avery Bradley       0       PG   25   2-Jun   
Brooklyn Nets                Bojan Bogdanovic      44       SG   27   8-Jun   
Charlotte Hornets               Nicolas Batum       5       SG   27   8-Jun   
Chicago Bulls                Cameron Bairstow      41       PF   25   9-Jun   
Cleveland Cavaliers       Matthew Dellavedova       8       PG   25   4-Jun   
Dallas Mavericks              Justin Anderson       1       SG   22   6-Jun   
Denver Nuggets                 Darrell Arthur       0       PF   28   9-Jun   
Detroit Pistons                  Joel Anthony      50        C   33   9-Jun   
Golden State Warriors         Leandro Barbosa      19       SG   33   3-Jun   
Houston Rockets                  Trevor Ariza       1       SF   30   8-Jun   
Indiana Pacers                    Lavoy Allen       5       PF   27   9-Jun   
Los Angeles Clippers             Cole Aldrich      45        C   27  11-Jun   
Los Angeles Lakers               Brandon Bass       2       PF   31   8-Jun   
Memphis Grizzlies                Jordan Adams       3       SG   21   5-Jun   
Miami Heat                         Chris Bosh       1       PF   32  11-Jun   
Milwaukee Bucks         Giannis Antetokounmpo      34       SF   21  11-Jun   
Minnesota Timberwolves        Nemanja Bjelica      88       PF   28  10-Jun   
New Orleans Pelicans            Alexis Ajinca      42        C   28   2-Jul   
New York Knicks                 Arron Afflalo       4       SG   30   5-Jun   
Oklahoma City Thunder            Steven Adams      12        C   22  Jul-00   
Orlando Magic                  Dewayne Dedmon       3        C   26  Jul-00   
Philadelphia 76ers                Elton Brand      42       PF   37   9-Jun   
Phoenix Suns                     Eric Bledsoe       2       PG   26   1-Jun   
Portland Trail Blazers        Cliff Alexander      34       PF   20   8-Jun   
Sacramento Kings                   Quincy Acy      13       SF   25   7-Jun   
San Antonio Spurs           LaMarcus Aldridge      12       PF   30  11-Jun   
Toronto Raptors               Bismack Biyombo       8        C   23   9-Jun   
Utah Jazz                       Trevor Booker      33       PF   28   8-Jun   
Washington Wizards              Alan Anderson       6       SG   33   6-Jun   

                        Weight                College      Salary  
Team                                                               
Atlanta Hawks              201           Old Dominion   2000000.0  
Boston Celtics             180                  Texas   7730337.0  
Brooklyn Nets              216         Oklahoma State   3425510.0  
Charlotte Hornets          200  Virginia Commonwealth  13125306.0  
Chicago Bulls              250             New Mexico    845059.0  
Cleveland Cavaliers        198           Saint Mary's   1147276.0  
Dallas Mavericks           228               Virginia   1449000.0  
Denver Nuggets             235                 Kansas   2814000.0  
Detroit Pistons            245                   UNLV   2500000.0  
Golden State Warriors      194         North Carolina   2500000.0  
Houston Rockets            215                   UCLA   8193030.0  
Indiana Pacers             255                 Temple   4050000.0  
Los Angeles Clippers       250                 Kansas   1100602.0  
Los Angeles Lakers         250                    LSU   3000000.0  
Memphis Grizzlies          209                   UCLA   1404600.0  
Miami Heat                 235           Georgia Tech  22192730.0  
Milwaukee Bucks            222                Arizona   1953960.0  
Minnesota Timberwolves     240             Louisville   3950001.0  
New Orleans Pelicans       248             California   4389607.0  
New York Knicks            210                   UCLA   8000000.0  
Oklahoma City Th

After grouping all the different team names, I then want to group all the inforation for one team. I chose to group all the information for Miami Heat.

In [28]:
#finding the values contained in the "portland Miami Heat" group
gk.get_group('Miami Heat')

Age                College  Height               Name  Number Position  \
339   32           Georgia Tech  11-Jun         Chris Bosh       1       PF   
340   31                   Duke   9-Jun          Luol Deng       9       SF   
341   30                    NaN   3-Jun       Goran Dragic       7       PG   
342   30                    NaN   7-Jun       Gerald Green      14       SF   
343   36                Florida   8-Jun      Udonis Haslem      40       PF   
344   34               Arkansas   7-Jun        Joe Johnson       2       SF   
345   24           Fresno State   4-Jun      Tyler Johnson       8       SG   
346   29                   Duke  10-Jun     Josh McRoberts       4       PF   
347   22              Tennessee   6-Jun    Josh Richardson       0       SG   
348   33                    NaN  10-Jun  Amar'e Stoudemire       5       PF   
349   34              Marquette   4-Jun        Dwyane Wade       3       SG   
350   23  Virginia Commonwealth   2-Jun      Briante Weber      12       PG   
351   26               Marshall  Jul-00   Hassan Whiteside      21        C   
352   20                   Duke   7-Jun    Justise Winslow      20       SF   
353   30                    NaN   9-Jun      Dorell Wright      11       SF   

         Salary  Weight  
339  22192730.0     235  
340  10151612.0     220  
341  14783000.0     190  
342    947276.0     205  
343   2854940.0     235  
344    261894.0     240  
345    845059.0     186  
346   5543725.0     240  
347    525093.0     200  
348    947276.0     245  
349  20000000.0     220  
350         NaN     165  
351    981348.0     265  
352   2481720.0     225  
353         NaN     205

After grabbing all of that data, I want to use describe again, but specifaclly for the team Miami Heat.

In [29]:
gk.get_group('Miami Heat').describe()

/Users/ziahgraceolivadaily/anaconda/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


Age     Number        Salary     Weight
count  15.000000  15.000000  1.300000e+01   15.00000
mean   28.933333  10.466667  6.347359e+06  218.40000
std     4.861902  10.377632  7.848628e+06   26.48126
min    20.000000   0.000000  2.618940e+05  165.00000
25%    25.000000   3.500000           NaN  202.50000
50%    30.000000   8.000000           NaN  220.00000
75%    32.500000  13.000000           NaN  237.50000
max    36.000000  40.000000  2.219273e+07  265.00000

Simlar to the code above I want to grab our best frined info, for the Miami Heat team.

In [30]:
gk.get_group('Miami Heat').info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 15 entries, 339 to 353
Data columns (total 8 columns):
Age         15 non-null int64
College     11 non-null object
Height      15 non-null object
Name        15 non-null object
Number      15 non-null int64
Position    15 non-null object
Salary      13 non-null float64
Weight      15 non-null int64
dtypes: float64(1), int64(3), object(4)
memory usage: 1.1+ KB


## Data analysis and conclusion

### Conclusion

By using the pandas function and then the groupby function I was able to get some interesting information on the Miami heat team. On the Team, the maximum age for the team is 36 years old. This leads me to believe that in your late 30s you are passed your professional basketbll days. Also, learning the maximum makes me wonder what the minimum age is. Looking back to the decribe function I can see that the minimum age for the miami heat is 20 years old. The mean age for the team is around 28 years old.

In additon to looking at the minimum, maximum, and mean age, I was able to look at the minimum, maximum, and mean salary for the Miami Heats. The minimum salary is $261,894.The maximum salary is $22,192,730. Lastly,the Mean salary is $6,347,359. This raises some questions (shown down below) and leads me to believe that their is a wide range of salaries in this baketball team, some making as little as two hundred and sixty one dollars, and some making as much as twenty two million.


### Questions Raised

What is/was the oldest professional basketball player? What is/was the youngest?

What is/was the highest salary for a professional basketball player? What is/was the lowest?

What leads one basketball player to making more money than another?




## Aknowledgements

Thank you to Navya, June, and Ms. Sconyers for the helping in lab 3.